src/index.js

```diff
@@ -0,0 +1,188 @@
+    removeButton.addEventListener('click', () => {
+      removeTask(li);
+
+      // window.location.reload();
+    });
+
+    const editButton = li.querySelector('.edit-button');
+    editButton.addEventListener('click', () => {
+      editTask(li);
+    });
+  });
+}
+
+renderTasks();
+
+function addTask() {
+  const taskValue = taskInput.value.trim();
+  // window.location.reload();
+
+  if (taskValue === '') {
+    return;
+  }
+
+  const task = {
+    text: taskValue,
+    completed: false,
+    index: savedTasks.length + 1,
+  };
+
+  savedTasks.push(task);
+  saveTasks();
+
+  const li = document.createElement('li');
+  const checkbox = document.createElement('input');
+  li.style.listStyle = 'none';
+  checkbox.type = 'checkbox';
+  checkbox.dataset.index = task.index - 1;
+  li.appendChild(checkbox);
+
+  li.innerHTML += `${task.text} <button class="remove-button">X</button> <button class="edit-button">Edit</button>`;
+  todoList.appendChild(li);
+
+  const removeButton = li.querySelector('.remove-button');
+  removeButton.addEventListener('click', () => {
+    removeTask(li);
+  });
+
+  const editButton = li.querySelector('.edit-button');
+
+  editButton.addEventListener('click', () => {
+    editTask(li);
+  });
+
+  taskInput.value = '';
+}
+
+// complete button
+
+// remove task
+
+// newly comein
+function saveTaskCompletedState(taskElement, isCompleted) {
+  const taskId = taskElement.getAttribute('id');
+  const taskData = {
+    completed: isCompleted,
+  };
+  localStorage.setItem(taskId, JSON.stringify(taskData));
+}
+
+function markTaskAsCompleted(taskElement) {
+  taskElement.classList.add('completed');
+  const checkbox = taskElement.querySelector('input[type="checkbox"]');
+  checkbox.checked = true;
+  saveTaskCompletedState(taskElement, true);
+}
+
+function markTaskAsUncompleted(taskElement) {
+  taskElement.classList.remove('completed');
+  const checkbox = taskElement.querySelector('input[type="checkbox"]');
+  checkbox.checked = false;
+  saveTaskCompletedState(taskElement, false);
+}
+
+function removeCompletedTasks() {
+  const checkboxes = document.querySelectorAll('input[type="checkbox"]');
+  checkboxes.forEach((checkbox) => {
+    checkbox.addEventListener('change', function () {
+      if (this.checked) {
+        const li = this.parentNode;
+        if (li.classList.contains('completed')) {
+          markTaskAsUncompleted(li);
+        } else {
+          markTaskAsCompleted(li);
+        }
+      } else {
+        const li = this.parentNode;
+        if (li.classList.contains('completed')) {
+          markTaskAsUncompleted(li);
+        } else {
+          saveTaskCompletedState(li, false);
+        }
+      }
+    });
+    if (checkbox.checked) {
+      const li = checkbox.parentNode;
+      markTaskAsCompleted(li);
+    }
+  });
+}
+
+// call removeCompletedTasks to initialize
+removeCompletedTasks();
+
+addButton.addEventListener('click', addTask);
+
+// completeButton();
```

Comment:
Please make sure you remove unused code from your code. This would allow you to write clean code.